### ISSUE TITLE
All if/else branches being entered

### ISSUE THREAD

**Issue by KLozes** [yuanming-hu/taichi#281]
I am running into a bug where all if/else branches are being entered by the program no matter what. For example, the program below should obviously print out 10000 for every element of x. But it instead prints 0 through 15 as if the "else" statement was entered after the "if". 

```python
import taichi as ti

ti.cfg.arch = ti.x86_64
#ti.cfg.arch = ti.cuda

x = ti.var(ti.f32)

@ti.layout
def xy():
  ti.root.dense(ti.i,16).place(x)

@ti.kernel
def if_test():
  for i in x:
    if i < 100:
      x[i] = 10000
    else:
      x[i] = i

if_test()

for i in range(16):
  print(x[i])
```

**Comment by yuanming-hu** [comment]
Thanks for reporting this! I'm looking into this issue.

**Comment by yuanming-hu** [comment]
Hi @KLozes 

This is a bug recently introduced when migrating to the LLVM backend. Now it is fixed in version v0.1.4. Thanks!

-Yuanming
